controllers/component_dependency_update_controller_test.go

```diff
@@ -325,49 +409,90 @@ var _ = Describe("Component nudge controller", func() {
 			}
 			pr.Status.CompletionTime = &metav1.Time{Time: time.Now()}
 			Expect(k8sClient.Status().Update(ctx, pr)).Should(BeNil())
+
 			Eventually(func() bool {
+				// check that multiple nudgeerror event were reported
 				events := v1.EventList{}
 				err := k8sClient.List(context.TODO(), &events)
 				Expect(err).ToNot(HaveOccurred())
-				op1nudged := false
-				op2nudged := false
 				failureCount := 0
 				for _, i := range events.Items {
-					if i.Reason == ComponentNudgedEventType && strings.Contains(i.Message, "quay.io.foo/bar:latest@sha256:12345") {
-						if strings.Contains(i.Message, "operator1") {
-							op1nudged = true
-						}
-						if strings.Contains(i.Message, "operator2") {
-							op2nudged = true
-						}
-					} else if i.Reason == ComponentNudgeFailedEventType {
+					if i.Reason == ComponentNudgeFailedEventType {
 						failureCount++
 					}
 				}
-				return !op1nudged && !op2nudged && failureCount == 3
+				// check that no renovate pipeline run was created
+				renovatePipelinesCreated := 0
+				prList := &tektonapi.PipelineRunList{}
+				err = k8sClient.List(context.TODO(), prList, &client.ListOptions{Namespace: UserNamespace})
+				Expect(err).ToNot(HaveOccurred())
+				for _, pipelineRun := range prList.Items {
+					if strings.HasPrefix(pipelineRun.ObjectMeta.Name, "renovate-pipeline") {
+						log.Info(pipelineRun.ObjectMeta.Name)
+						renovatePipelinesCreated += 1
+					}
+				}
+
+				return renovatePipelinesCreated == 0 && failureCount == 3
 			}, timeout, interval).WithTimeout(ensureTimeout).Should(BeTrue())
 		})
 	})
 
 	Context("Test renovate config template", func() {
-		// This test mostly just makes sure the template generates sane config
 		It("test template output", func() {
-
+			componentName := "test-component"
 			component := &applicationapi.Component{}
-			component.Name = "test-component"
+			component.Name = componentName
```

Comment:
It looks like a unit test

Reply:
we had that like in current tests

if we create more unit tests we can move it to separate file

for this one test, I am going to leave it here